classes/migration/upgrade/v3_4_0/I7366_UpdateUserAPIKeySettings.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+namespace PKP\migration\upgrade\v3_4_0;
+
+use Illuminate\Support\Facades\DB;
+use PKP\install\DowngradeNotSupportedException;
+
+class I7366_UpdateUserAPIKeySettings extends \PKP\migration\Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        $userSettings = DB::table('user_settings')
+            ->select(['user_id', 'setting_name'])
+            ->where('setting_name', 'apiKeyEnabled')
+            ->distinct()
+            ->get();
+        
+        $userSettingsWithApiKey = DB::table('user_settings')
+                ->select(['user_id', 'setting_name'])
+                ->where('user_id', $userSettings->pluck('user_id')->toArray())
+                ->where('setting_name', 'apiKey')
+                ->distinct()
+                ->get();
+        
+        DB::table('user_settings')
+            ->whereIn(
+                'user_id', 
+                $userSettings
+                    ->pluck('user_id')
+                    ->diff($userSettingsWithApiKey->pluck('user_id'))
+                    ->toArray()
+            )
+            ->where('setting_name', 'apiKeyEnabled')
+            ->delete();
+    }
+
+    /**
+     * Reverse the migration.
+     */
+    public function down(): void
+    {
+        throw new DowngradeNotSupportedException('Downgrade unsupported due to data update');
```

Comment:
I'd recommend leaving the `down()` function empty, rather than throwing an exception. This class just deletes invalid data, so while a 100% proper `down()` implementation would reconstruct the bad data, that's more trouble than it's worth -- and I don't expect negative impacts to just leaving the bad data deleted.

Reply:
Done.